Support multiple -*notify commands

luke-jr opened the pull request:
Useful, intuitive, and avoids creating issues with #22354

kristapsk commented on src/wallet/wallet.cpp:
```diff
@@ -939,29 +939,31 @@ CWalletTx* CWallet::AddToWallet(CTransactionRef tx, const CWalletTx::Confirmatio
-        // https://github.com/bitcoin/bitcoin/pull/13339#issuecomment-461288094
-        boost::replace_all(strCmd, "%w", ShellEscape(GetName()));
+    if (gArgs.IsArgSet("-walletnotify")) {
+#ifdef WIN32
```
Comment below `#ifndef WIN32` from previous code about why it's not supported currently should be copied here IMO.

luke-jr replied:
Copied

kristapsk commented:
Concept ACK

benthecarman commented:
Concept ACK

benthecarman commented:
Looks like this doesn't touch -startupnotify, does that support multiple already?

luke-jr commented:
Whoops, fixed.

MarcoFalke commented on src/init.cpp:
```diff
@@ -683,9 +683,8 @@ static void CleanupBlockRevFiles()
-    std::string cmd = args.GetArg("-startupnotify", "");
-    if (!cmd.empty()) {
-        std::thread t(runCommand, cmd);
+    for (std::string command : gArgs.GetArgs("-startupnotify")) {
```
```suggestion
    for (std::string cmd : args.GetArgs("-startupnotify")) {
```

Seems odd to use gArgs when args is passed in. Also, why change the symbol name to increase the diff?

luke-jr replied:
gArgs: Fixed.

Symbol name: It doesn't increase the diff to adjust it to our current style.

luke-jr committed:
scripted-diff: test: rename `strCmd` to `command`

-BEGIN VERIFY SCRIPT-
sed -i 's/\bstrCmd\b/command/g' $(git grep -l strCmd)
-END VERIFY SCRIPT-

luke-jr committed:
Support multiple -*notify commands

luke-jr commented:
Rebased and addressed review comments

achow101 commented:
Closing this as it has not had any activity in a while. If you are interested in continuing work on this, please leave a comment so that it can be reopened.

kristapsk commented:
Guess this was closed just because of no rebase after May 6 and lack of ACKs? Seems useful in context of #25975.

luke-jr commented:
Feel free to reopen. As with most of my PRs, I continue to maintain it for Knots.
